irohad/ametsuchi/impl/mutable_storage_impl.cpp

```diff
@@ -23,48 +22,75 @@ namespace iroha {
         std::shared_ptr<shared_model::interface::CommonObjectsFactory> factory)
         : top_hash_(top_hash),
           sql_(std::move(sql)),
-          wsv_(std::make_shared<PostgresWsvQuery>(*sql_, factory)),
+          peer_query_(std::make_unique<PeerQueryWsv>(
+              std::make_shared<PostgresWsvQuery>(*sql_, factory))),
           block_index_(std::make_unique<PostgresBlockIndex>(*sql_)),
           command_executor_(std::make_shared<PostgresCommandExecutor>(*sql_)),
           committed(false),
           log_(logger::log("MutableStorage")) {
       *sql_ << "BEGIN";
     }
 
-    bool MutableStorageImpl::check(
-        const shared_model::interface::Block &block,
-        MutableStoragePredicate predicate) {
-      PeerQueryWsv peer_query(wsv_);
-      return predicate(block, peer_query, top_hash_);
-    }
-
-    bool MutableStorageImpl::apply(const shared_model::interface::Block &block) {
+    bool MutableStorageImpl::apply(const shared_model::interface::Block &block,
+                                   MutableStoragePredicate function) {
```

Comment:
Maybe keep name `predicate` instead of `function`?

Reply:
It will make it inconsistent with TemporaryWsv, because `function` is used there.

Reply:
Then it can be made `precidate` in `TemporaryWsv` as well? This improves code understanding, I think